code/modules/mining/machinery/material_extractor.dm

```diff
@@ -0,0 +1,408 @@
+	return turn(dir, -90)
+
+/obj/machinery/atmospherics/unary/material/extractor/Initialize(mapload, d = 0, populate_parts = TRUE)
+	. = ..()
+	if(populate_parts)
+		output_container = new/obj/item/chems/glass/bucket(src)
+	air_contents.volume = GAS_EXTRACTOR_GAS_TANK
+	if(!reagents)
+		create_reagents(GAS_EXTRACTOR_REAGENTS_TANK)
+	if(!input_buffer)
+		input_buffer = new(src)
+		input_buffer.create_reagents(GAS_EXTRACTOR_REAGENTS_INPUT_TANK) //Did this here because reimplementing that in the new() proc failed a test for some reasons
+	verbs |= /obj/machinery/atmospherics/unary/material/extractor/verb/FlushReagents
+	verbs |= /obj/machinery/atmospherics/unary/material/extractor/verb/FlushGas
+	QUEUE_TEMPERATURE_ATOMS(src)
+
+/obj/machinery/atmospherics/unary/material/extractor/Destroy()
+	if(istype(input_buffer))
+		qdel(input_buffer)
+	. = ..()
+
+/obj/machinery/atmospherics/unary/material/extractor/Bumped(var/obj/O)
+	if(QDELETED(O)) //Because we qdel object at the input if we can process them. And its possible this might happen 
+		return
+	//We only override for entities touching the machine from the input's direction only
+	if(get_dir(loc, O.loc) != get_input_dir() || inoperable() || !O.checkpass(PASS_FLAG_TABLE) || O.anchored)
+		return ..()
+	
+	//2 possible cases here. One we got something that we can turn into liquids or gas (with or without accompanying solid reagent at STP) 
+	// OR we get something that only contains matter that's solid at STP, which we should just pass along so whatever else is in the 
+	// conveyor line can process it.
+	if(can_process_object(O))
+		if(calc_resulting_reagents_total_vol(O) > input_tank_free_volume())
+			return //If we can process it, but there's no room currently in the input, just don't interact with it for now!
+		process_ore(O)
+	else
+		O.dropInto(get_output_loc())
+
+/obj/machinery/atmospherics/unary/material/extractor/ProcessAtomTemperature()
+	if(operable() && use_power)
+		//We process temp for the input reagent_holder too
+		input_buffer.temperature = GAS_EXTRACTOR_OPERATING_TEMP
+		temperature = GAS_EXTRACTOR_OPERATING_TEMP
+		return TRUE
+	return ..()
+
+/obj/machinery/atmospherics/unary/material/extractor/examine(var/mob/user)
+	. = ..()
+	//Only display info if the screen is there
+	if(get_component_of_type(/obj/item/stock_parts/console_screen))
+		to_chat(user, SPAN_NOTICE("The processing tank gauge reads [round(input_buffer.reagents.total_volume)]/[input_buffer.reagents.maximum_volume] units of liquid"))
+		to_chat(user, SPAN_NOTICE("The internal storage tank gauge reads [round(reagents.total_volume)]/[reagents.maximum_volume] units of liquid"))
+		to_chat(user, SPAN_NOTICE("The internal gas tank pressure gauge reads [air_contents.return_pressure()] kPa"))
+
+		if(is_output_container_full() || is_internal_tank_full())
+			to_chat(user, SPAN_WARNING("Currently idling because one or more of its liquid tanks are full.."))
```

Comment:
Double full stop, also could probably use 'It is currently' as an opener.

Reply:
Not sure what you mean here? Double full stop? And will do for the flavor text fix.

Reply:
You have two full stops at the end of the string is what I meant.

Reply:
Oh I see it now! Sorry.